packages/flutter/lib/src/painting/text_span.dart

```diff
@@ -304,27 +308,32 @@ class TextSpan extends DiagnosticableTree {
   /// Describe the difference between this text span and another, in terms of
   /// how much damage it will make to the rendering. The comparison is deep.
   ///
+  /// Comparing a [TextSpan] with a [WidgetSpan] will result in [RenderComparison.layout].
+  ///
   /// See also:
   ///
   ///  * [TextStyle.compareTo], which does the same thing for [TextStyle]s.
-  RenderComparison compareTo(TextSpan other) {
+  RenderComparison compareTo(InlineSpan other) {
     if (identical(this, other))
       return RenderComparison.identical;
-    if (other.text != text ||
-        children?.length != other.children?.length ||
-        (style == null) != (other.style == null))
+    if (!(other is TextSpan))
```

Comment:
prefer using `...is!...` over `!(...is...)`

Reply:
though i guess we really should just compare runtimeTypes rather than use `is`, because what if someone subclasses TextSpan?